hector_gamepad_manager/src/hector_gamepad_manager.cpp

```diff
@@ -29,8 +29,11 @@ HectorGamepadManager::HectorGamepadManager( const rclcpp::Node::SharedPtr &node
   // load meta switch config and all referenced config files
   if ( loadConfigSwitchesConfig( config_switches_filename ) ) {
     switchConfig( default_config_ );
+
+    rclcpp::SubscriptionOptions options;
+    options.callback_group = node->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
     joy_subscription_ = ocs_ns_node_->create_subscription<sensor_msgs::msg::Joy>(
-        "joy", 1, std::bind( &HectorGamepadManager::joyCallback, this, std::placeholders::_1 ) );
+        "joy", 1, std::bind( &HectorGamepadManager::joyCallback, this, std::placeholders::_1 ), options);
```

Comment:
This is not needed if we are using a single threaded executor (which we should).